cores/esp32/esp32-hal-uart.c

```diff
@@ -326,6 +326,34 @@ uint32_t uartAvailableForWrite(uart_t* uart)
     return available;
 }
 
+size_t uartReadBytes(uart_t* uart, uint8_t *buffer, size_t size, uint32_t timeout_ms)
+{
+    if(uart == NULL || size == 0 || buffer == NULL) {
+        return 0;
+    }
+
+    size_t bytes_read = 0;
+
+    UART_MUTEX_LOCK();
+
+    if (uart->has_peek) {
+        uart->has_peek = false;
+        *buffer++ = uart->peek_byte;
+        size--;
+        bytes_read = 1;
+    }
+
+    if (size > 0) {
+       int len = uart_read_bytes(uart->num, buffer, size, pdMS_TO_TICKS(timeout_ms));
```

Comment:
Since you're calling it with 0 msec timeout, I wonder what does 0 timeout mean here?
Is it "no wait" or "wait forever" ?
It isn't defined here: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html

Reply:
Zero means, no waiting. If the bytes are not available, it will not wait and return immediately.
portMAX_DELAY is used as the "waiting for ever".

Reply:
thanks for the update!